contracts/MerkleAirdrop.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+import "./MerkleProof.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/structs/BitMaps.sol";
+
+/**
+ * @dev A contract to allow users to claim tokens via a 'merkle airdrop'.
+ */
+contract MerkleAirdrop is Ownable {
+    using BitMaps for BitMaps.BitMap;
+
+    address public immutable sender;
+    IERC20 public immutable token;
+    bytes32 public immutable merkleRoot;
+    BitMaps.BitMap private claimed;
+
+    event MerkleRootChanged(bytes32 merkleRoot);
+    event Claim(address indexed claimant, uint256 amount);
+
+    /**
+     * @dev Constructor.
+     * @param _sender The account to send airdrop tokens from.
+     * @param _token The token contract to send tokens with.
+     */
+    constructor(address _sender, IERC20 _token, bytes32 _merkleRoot) {
+        sender = _sender;
+        token = _token;
+        merkleRoot = _merkleRoot;
+    }
+
+    /**
+     * @dev Claims airdropped tokens.
+     * @param recipient The account being claimed for.
+     * @param amount The amount of the claim being made.
+     * @param merkleProof A merkle proof proving the claim is valid.
+     */
+    function claimTokens(address recipient, uint256 amount, bytes32[] calldata merkleProof) external {
```

Comment:
Is it better to use msg.sender and not use address recipient ?
It's extremely unlikely that anyone other than the actual address to call claim tokens ( and spend 
unnecessary pay gas fee !) but isn't better to have check 
if recipient is the actual claiming address or not ?

Reply:
There's no harm in allowing people to claim for other accounts, and some people will find it useful.

Reply:
The only scenario is where the user don't  want to claim the drop , maybe due to tax implications and someone claims it on user behalf unknowingly.